lib/graphiti/adapters/abstract.rb

```diff
@@ -400,6 +400,13 @@ def destroy(model_instance)
         raise "you must override #destroy in an adapter subclass"
       end
 
+      def apply_includes_on_scope(scope, sideloads)
```

Comment:
```suggestion
      def before_resolve(scope, query)
```
Override `Graphiti::Resource#before_resolve` to delegate to adapter. And move the logic from `Scope` to our adapter.